src/components/button-swap/button-swap.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Helper to get the button from the supplied wrapper element
+ */
+const getButton = (wrapper: HTMLElement) =>
+  wrapper.querySelector('.js-button-swap__button') as HTMLButtonElement;
+
+/**
+ * Helper to get the status message from the supplied wrapper element
+ */
+const getStatusMessage = (wrapper: HTMLElement) =>
+  wrapper.querySelector('.js-button-swap__message') as HTMLElement;
+
+/**
+ * Performs click actions
+ */
+const onClick = ({
+  event,
+  hiddenWrapper,
+  visibleWrapper,
+  callback,
```

Comment:
Nit: make this required and remove the if statement below?

Reply:
Sure! 👍

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1516/commits/a7a49c435ee8d4b801cecf5b133241f520a3b1b5